File: LAB 4/Scan-Converting_a_Rectangle.m
clc; clear; close all;

X1 = 10; Y1 = 10;
X2 = 40; Y2 = 30;

figure;
hold on;
grid on;
axis equal;
title('Rectangle Scan Conversion');
xlabel('X-axis');
ylabel('Y-axis');

X_VALS = [];
Y_VALS = [];

for X = X1:X2
    X_VALS = [X_VALS X X];
    Y_VALS = [Y_VALS Y1 Y2];
end

for Y = Y1+1:Y2-1
    X_VALS = [X_VALS X1 X2];
    Y_VALS = [Y_VALS Y Y];
end

plot(X_VALS, Y_VALS, 'ks', 'MarkerSize', 4, 'MarkerFaceColor', 'k');
plot([X1 X2 X2 X1 X1], [Y1 Y1 Y2 Y2 Y1], 'r-', 'LineWidth', 2);

File: LAB 4/Python/Scan-Converting_a_Rectangle.py
import matplotlib.pyplot as plt

X1, Y1 = 10, 10
X2, Y2 = 40, 30

plt.figure()
plt.grid(True)
plt.axis('equal')
plt.title('Rectangle Scan Conversion')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')

X_VALS = []
Y_VALS = []

for X in range(X1, X2 + 1):
    X_VALS.append(X)
    X_VALS.append(X)
    Y_VALS.append(Y1)
    Y_VALS.append(Y2)

for Y in range(Y1 + 1, Y2):
    X_VALS.append(X1)
    X_VALS.append(X2)
    Y_VALS.append(Y)
    Y_VALS.append(Y)

plt.plot(X_VALS, Y_VALS, 'ks', markersize=4, markerfacecolor='k')
plt.plot([X1, X2, X2, X1, X1], [Y1, Y1, Y2, Y2, Y1], 'r-', linewidth=2)
plt.show()
